File: p0291simpleactivityresult/src/main/java/com/dovhal/p0291simpleactivityresult/MainActivity.java
package com.dovhal.p0291simpleactivityresult;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    TextView tvName;
    Button inputNameBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvName = findViewById(R.id.tvName);
        inputNameBtn = findViewById(R.id.inputNameBtn);

        inputNameBtn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent intent = new Intent(this, InputActivity.class);
        startActivityForResult(intent, 1);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (data == null) { return; }
        String name = data.getStringExtra("name");
        tvName.setText("Your name is: " + name);
    }
}

File: k001hb/src/main/java/com/dovhal/k001hb/MainActivity.kt
package com.dovhal.k001hb

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tv.text = "Hello from Kotlin"
        var flag = true;

        btnPush.setOnClickListener(){
            flag = !flag
            if (flag){
                btnPush.text = "Pushed"
            } else btnPush.text = "Not pushed"
        }
    }
}
